### PR TITLE
fix(450): Allow pipeline to be removed if repository does not exist

### PR DESCRIPTION
**Pull request by Filbird**
## Context
Currently, pipelines cannot be removed if its underlying repository does not exist.

## Objective
Allow pipeline to be removed if `getPermissions` returns a 404 and only if `privateRepo` is disabled.

## References
Pending:
  - https://github.com/screwdriver-cd/scm-bitbucket/pull/52
https://github.com/screwdriver-cd/screwdriver/issues/450

### CODE REVIEW

File: plugins/pipelines/remove.js [hunk 1]
```diff
@@ -52,6 +52,12 @@ module.exports = () => ({
                                 + 'does not have admin permission for this repo');
                         }
                     })
+                    .catch((error) => {
+                        // Allow pipeline to be removed if the repository does not exist
+                        if (error.code !== 404) {
```
[the remark this answers is not in the document]
Comment (Filbird) [reply plugins/pipelines/remove.js]: Yep, added the check!

Will look into the Bitbucket and Gitlab cases

**Comment by tkyi** [comment]
It's possible we could get a 404 code for a different reason, and I'm concerned we're not covering all cases right now. What if we don't allow users to delete a pipeline when there's a 404, but we allow Screwdriver cluster admins. That way, users can file a support request if they really need the pipeline to be deleted and we can manually do it.
We can reuse the check used here: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/banners/index.js#L24

**Comment by Filbird** [comment]
@tkyi  I see, this would certainly be a safer way of solving the issue!

I'll post your comment on the issue page and update this PR.

File: plugins/pipelines/remove.js [hunk 2]
```diff
@@ -52,6 +55,18 @@ module.exports = () => ({
                                 + 'does not have admin permission for this repo');
                         }
                     })
+                    .catch((error) => {
+                        // Lookup whether user is admin
+                        const adminDetails = request.server.plugins.pipelines
```
Comment (tkyi) [inline plugins/pipelines/remove.js]: I think you can reuse the banners code like this instead: https://github.com/screwdriver-cd/screwdriver/blob/master/plugins/builds/update.js#L67
Not sure what makes more sense

Comment (Filbird) [reply plugins/pipelines/remove.js]: Makes sense! Better than duplicating it.

**Pull request merged**